Unity/BananaClient/TcpAdapter.cs

```diff
@@ -155,25 +163,43 @@ private int writePrimitiveBytes(byte[] dest, int offset, ulong data)
         }
 
         private void Receiving() {
-            try {
+            try 
+            {
                 // Waiting for sendingThread to establish the connnection.
                 while (running && error == "" && networkStream == null)
                     Thread.Sleep(0);
 
+                ushort eventType = 0;
                 while (running && error == "") {
-                    ushort eventType = ReadUInt16(networkStream);
-                   
-                    if (eventType == NetworkMessage.EventType_ConnectServer) {
+                    try
+                    {
+                        eventType = ReadUInt16(networkStream);
+                    }
+                    catch (Exception)
+                    {
+                        Thread.Sleep(100);
+                        continue;
+                    }
+                    
+                    if (eventType == NetworkMessage.EventType_ConnectServer) 
+                    {
                         ConnectServerResponse response;
                         response.eventType = eventType;
                         response.sessionId = ReadBytes(networkStream, 20);
                         response.userId = ReadUInt64(networkStream);
                         response.errorCode = ReadByte(networkStream);
+                        Byte[] buf = new byte[520-31];
+                        networkStream.Read(buf, 0, buf.Length); // dump redundant bytes
```

Comment:
According to the doc, error code is the last field that is expected from the response (https://ibb.co/SmxmdPL). There is no redundant bytes. 

Here is the doc: https://docs.google.com/document/d/1lKwLcgIwcYABAs81AN1WuSWuFH0Nocdk18n5Ikb4EbM/edit

Reply:
The dumping is to ensure the whole 520 bytes are read from the stream, this is not applicable for the actual use case as the server is expected to return a proper response. We can look into a more sophisticate mock server object for testing purpose.

Reply:
there is no 520 bytes expected from this `EventType_ConnectServer` from the server (https://ibb.co/SmxmdPL). isn't there?

Reply:
I concur; The current implementation read from the same memorystream to get the response. The mock class for Unit test should form a proper response before returning to the read of the receiving method.